File: src/main/java/pcb/units/concrete/physics/OunceUnit.java
package pcb.units.concrete.physics;

import pcb.units.base.BaseUnit;
import pcb.units.dimensions.physics.MassUnit;

import java.math.BigDecimal;

import static java.math.MathContext.DECIMAL64;

public class OunceUnit
		extends BaseUnit<MassUnit>
		implements MassUnit {

	public static final OunceUnit instance = new OunceUnit();

	private static final BigDecimal GRAMS_PER_OUNCE = new BigDecimal("28.349523125");

	private OunceUnit() {
		super("oz", "ounce", "ounces",
				value -> value.multiply(GRAMS_PER_OUNCE, DECIMAL64),
				value -> value.divide(GRAMS_PER_OUNCE, DECIMAL64)
		);
	}
}

File: implementation/src/main/java/pcb/uwu/unit/base/OunceUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.unit.base.OunceUnit.CONVERT

object OunceUnit
    : MassUnit(symbol = "oz",
               singularName = "ounce",
               pluralName = "ounces",
               translationToCanonical = { it * CONVERT },
               translationFromCanonical = { it / CONVERT })
{
    const val CONVERT = 28.349523125
}
